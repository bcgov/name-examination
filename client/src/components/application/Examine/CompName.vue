/* eslint-disable */
<template>
  <div>
    <div class="name-sect">
      <div class="row">
        <div class="col">

          <div id="top-buttons">
            <button class="btn btn-sm btn-secondary" id="examine-get-next-button"
                    v-if="!is_making_decision && !is_my_current_nr"
                    @click="getNextCompany()" >Get Next</button>
            <button class="btn btn-sm btn-warning" id="examine-hold-button"
                    v-if="!is_making_decision && is_my_current_nr"
                    @click="holdRequest()">Hold</button>
            <button class="btn btn-sm btn-primary" id="examine-decide-button"
                    v-if="!is_making_decision && !is_complete && is_my_current_nr"
                    @click="startDecision()" >Approve/Reject...</button>

            <button class="btn btn-sm btn-primary" id="decision-approve-button"
                    v-if="is_making_decision" @click="nameAccept()">
              <span v-if="acceptance_will_be_conditional">Conditionally </span>Approve
            </button>
            <button class="btn btn-sm btn-danger" id="decision-reject-button"
                    v-if="is_making_decision" @click="nameReject()" >Reject</button>
            <button class="btn btn-sm btn-secondary" id="decision-cancel-button"
                    v-if="is_making_decision" @click="is_making_decision=false">Cancel</button>

            <button class="btn btn-sm btn-danger" id="examine-re-open-button"
                    v-if="is_complete && !is_furnished" @click="reOpen()" >
              Re-Open</button>

            <!-- EXAMINE button - to claim/examine an NR that is on hold -->
            <button class="btn btn-sm btn-primary" id="examine-button" v-if="can_claim"
                    @click="claimNR()" >Examine</button>
          </div>

          <table>
            <tr class="name-option"
                v-bind:class="{'active-name-option': currentChoice==1,
                               accepted: compName1.state == 'A'}">
              <td>1.</td>
              <td id="name1">
                {{ compName1.name }}
                <span class="name-state-icon" v-html="setIcon(compName1.state)"></span>
                <button class="btn btn-undo" v-if="is_undoable_1"
                        v-on:click="undoDecision(1)">Undo Decision</button>
                <span class="decision-text">{{ compName1.decision_text }}</span>
              </td>
            </tr>
            <tr class="name-option"
                v-bind:class="{'active-name-option': currentChoice==2,
                               accepted: compName2.state == 'A'}">
              <td>2.</td>
              <td id="name2">
                {{ compName2.name }}
                <span class="name-state-icon" v-html="setIcon(compName2.state)"></span>
                <button class="btn btn-undo" v-if="is_undoable_2"
                        v-on:click="undoDecision(2)">Undo Decision</button>
                <span class="decision-text">{{ compName2.decision_text }}</span>
              </td>
            </tr>
            <tr class="name-option"
                v-bind:class="{'active-name-option': currentChoice==3,
                               accepted: compName3.state == 'A'}">
              <td>3.</td>
              <td id="name3" >
                {{ compName3.name }}
                <span class="name-state-icon" v-html="setIcon(compName3.state)"></span>
                <button class="btn btn-undo" v-if="is_undoable_3"
                        v-on:click="undoDecision(3)">Undo Decision</button>
                <span class="decision-text">{{ compName3.decision_text }}</span>
              </td>
            </tr>
          </table>

          <div >
            <div align="right" v-if="!is_making_decision && !is_complete && is_my_current_nr">
              <button class="btn btn-sm btn-outline-primary" id="examine-quick-approve-button"
                      @click="quickApprove">Quick Approve</button>
              <button class="btn btn-sm btn-outline-danger" id="examine-reject-descriptive-button"
                      @click="rejectDescriptive">Reject Descriptive</button>
              <button class="btn btn-sm btn-outline-danger" id="examine-reject-distinctive-button"
                      @click="rejectDistinctive">Reject Distinctive</button>
            </div>
            <div v-if="!is_making_decision" id="manual-search">
              <form class="form-inline my-2 my-lg-0" @submit.prevent="onSubmit">
                <input class="search form-control" v-model="searchStr" />
                <button class="btn-search" type="submit"><i class="fa fa-search" /></button>
                <button class="btn-reset" v-if="is_running_manual_search" @click="resetSearchStr">
                  <i class="fa fa-times" /></button>
              </form>
            </div>
          </div>

        </div>

      </div>

    </div>
  </div>
</template>

<script>
/* eslint-disable */
  export default {
    name: 'CompName',
    data: function () {
      return {
        searchStr: '',
        retval: [],
        is_running_manual_search: false,
      }
    },
    computed: {
      decision_made: {
        get: function () {
          return this.$store.getters.decision_made;
        },
        set: function (value) {
          this.$store.commit('decision_made', value);
        }
      },
      currentState() {
        return this.$store.getters.currentState;
      },
      userId() {
        return this.$store.getters.userId;
      },
      is_my_current_nr() {
        return this.$store.getters.is_my_current_nr;
      },
      is_complete() {
        return this.$store.getters.is_complete;
      },
      is_furnished() {
        return this.$store.getters.furnished;
      },
      is_editing() {
        return  this.$store.getters.is_editing;
      },
      is_making_decision: {
        get: function() {
          return this.$store.getters.is_making_decision;
        },
        set: function(value) {
          this.$store.commit('is_making_decision', value);
        }
      },
      acceptance_will_be_conditional() {
        return this.$store.getters.acceptance_will_be_conditional;
      },
      can_claim() {
        console.log('got to can_claim with status ' + this.currentState);
        // can this user claim the NR? Based on state.
        if (['DRAFT', 'HOLD'].indexOf(this.currentState) > -1) return true;
        else return false;
      },
      compName1() {
        return this.$store.getters.compName1;
      },
      compName2() {
        return this.$store.getters.compName2;
      },
      compName3() {
        return this.$store.getters.compName3;
      },
      currentNameObj: {
        get: function() {
          return this.$store.getters.currentNameObj;
        },
        set: function (value) {
          this.$store.commit('currentNameObj', value);
        }
      },
      currentName() {
        return this.$store.getters.currentName;
      },
      currentChoice: {
        get: function () {
          return this.$store.getters.currentChoice
        },
        set: function (value) {
          this.$store.commit('currentChoice', value);
        }
      },
      is_undoable_1() {
        // first test generic reasons why a name would or wouldn't be undoable
        var undoable = this.is_undoable(this.compName1);

        if (undoable) {
          // if name choices 2 and 3 have not been decided, then 1 is undoable
          if ((this.compName2.state == 'NE' || this.compName2.state == null) &&
              (this.compName3.state == 'NE' || this.compName3.state == null)) {
            undoable = true;
          }
          else undoable = false;
        }

        return undoable;
      },
      is_undoable_2() {
        // first test generic reasons why a name would or wouldn't be undoable
        var undoable = this.is_undoable(this.compName2);

        if (undoable) {
          // if name choice 3 has not been decided, then 2 is undoable
          if (this.compName3.state == 'NE' || this.compName3.state == null) {
            undoable = true;
          }
          else undoable = false;
        }

        return undoable;
      },
      is_undoable_3() {
        // first test generic reasons why a name would or wouldn't be undoable
        var undoable = this.is_undoable(this.compName3);

        return undoable;
      },
      listDecisionReasons() {
        return this.$store.getters.listDecisionReasons;
      },
    },
    mounted() {
      console.log('Compname Mounted')
      if(this.$store.getters.nrNumber == null){
        console.log('Mounted->get next NR number')
        this.$store.dispatch('getpostgrescompNo');
      }
    },
    methods: {
      getNextCompany() {
        this.$store.dispatch('getpostgrescompNo');
      },
      startDecision() {
        this.$store.state.is_making_decision = true;
      },
      nameAccept() {
        this.$store.commit('decision_made', 'A');
      },
      nameReject() {
        this.$store.commit('decision_made', 'R');
      },
      reOpen() {
        this.$store.dispatch('updateNRState', 'INPROGRESS');
      },
      claimNR() {
        this.$store.dispatch('updateNRState', 'INPROGRESS');
      },
      holdRequest() {
        this.$store.dispatch('updateNRState', 'HOLD');
      },
      runRecipe(){
        this.$store.dispatch('runRecipe')
      },
      setIcon(name_state) {
        if (name_state == 'R') {
          return '<i class="fa fa-times icon-rejected"></i>';
        }
        else if (name_state == 'A') {
          return '<i class="fa fa-check icon-accepted"></i>';
        }
        else return '';
      },
      undoDecision(name_number) {
        console.log(name_number);
        this.$store.dispatch('undoDecision', name_number);

        // set the undone name choice to the current (actionable) choice
        if (name_number == 1) this.currentNameObj = this.compName1;
        if (name_number == 2) this.currentNameObj = this.compName2;
        if (name_number == 3) this.currentNameObj = this.compName3;

      },
      is_undoable(name) {

        // if the NR is closed in any way, a name is not undoable - the NR will have to be
        // re-opened first.
        if (this.currentState != 'INPROGRESS') return false;

        // if the NR is furnished, nothing is undoable
        if (this.$store.state.furnished)  return false;

        // if this name is complete (ie: anything other than NE) it's undoable
        if (name.state == 'NE' || name.state == null) return false;

        return true;
      },
      quickApprove() {
        var approvalStr = ''
        this.retval.push(approvalStr)
        console.log('quickApprove')

        this.decision_made = 'A'
       // this.nameAccept()
        this.nameAcceptReject()
      },
      rejectDescriptive() {
        // When this was written, the 13th index of listDecisionReasons was the string needed for a descriptive term missing
        // it was decided to HARD CODE this value until another solution is found
        // var descriptiveStr = this.listDecisionReasons[13].reason
        var descriptiveStr = 'Require descriptive second word or phrase * E.G. ' +
                             'Construction, Gardening, Investments, Holdings, Etc.'

        this.retval.push(descriptiveStr)
        this.decision_made = 'R'
        //this.nameReject()
        this.nameAcceptReject()
      },
      rejectDistinctive() {
        // When this was written, the 16th index of listDecisionReasons was the string needed for a distinctive term missing
        // it was decided to HARD CODE this value until another solution is found
        // var distinctiveStr = this.listDecisionReasons[16].reason
        var distinctiveStr = "Require distinctive, nondescriptive first word or prefix * E.G. " +
                             "Person's name, initials, geographic location, etc."
        this.retval.push(distinctiveStr)
        this.decision_made = 'R'
        //this.nameReject()
        this.nameAcceptReject()
      },
      onSubmit()
      {
        console.log("Running manual recipe");
        this.$store.dispatch('runManualRecipe', this.searchStr);

        if (this.searchStr != this.currentName) this.is_running_manual_search = true;
      },
      resetSearchStr(){
        this.searchStr = this.currentName
        this.is_running_manual_search = false;
      },
      nameAcceptReject() {

        // save decision text, state, and up to three conflicts
        //this.currentNameObj.decision_text = this.customer_message_display.substr(0,955);
        this.currentNameObj.decision_text = this.retval[0]

        console.log('nameAcceptReject decision_made:' + this.decision_made)
        if (this.decision_made == 'A') {
          this.currentNameObj.state = 'A'; // accepted

          // Not needed for FAST PATH Approval / Reject
          // conditionally accepted if any conditions selected with condition_required flag TRUE
          //for (let record in this.conditions_selected) {
            //if (record.consent_required) {
              //this.currentNameObj.state = 'C';
              //break;
            //}
          //}
        }
        else {
          this.currentNameObj.state = 'R';
        }
        // Not needed for FAST PATH Approval / Reject
        //for (var i = 0; i < this.conflicts_selected.length; i++) {
          //if (i == 0) {
            //this.currentNameObj.conflict1 = this.conflicts_selected[i].text;
            //this.currentNameObj.conflict1_num = this.conflicts_selected[i].nrNumber;
          //}
          //if (i == 1) {
            //this.currentNameObj.conflict2 = this.conflicts_selected[i].text;
            //this.currentNameObj.conflict2_num = this.conflicts_selected[i].nrNumber;
          //}
          //if (i == 2) {
            //this.currentNameObj.conflict3 = this.conflicts_selected[i].text;
            //this.currentNameObj.conflict3_num = this.conflicts_selected[i].nrNumber;
          //}
        //}

        // send decision to API and reset flags
        this.$store.dispatch('nameAcceptReject');
        this.decision_made = null;
        this.is_making_decision = false;
      },
    },
    watch: {
      currentName: function (val) {
        console.log('CompName.currentName watcher fired:' + val)
        this.searchStr =  val
      },
      currentChoice: function (val) {
        console.log('CompName.currentChoice watcher fired:' + val)
        if(val != undefined ) { this.runRecipe() }
      },
      nrNumber: function (val) {
        console.log('CompName.nrNumber watcher fired:' + val)
        if(val != null){ this.runRecipe()}
      }
    }
  }
</script>


<style scoped>
  .name-sect {
  }
  .name-option {
    font-size:1.2em;
    text-align: left;
  }
  .active-name-option{
    font-weight: bold;
  }

  #top-buttons {
    float: right;
    margin: 10px 0 10px 10px;
  }
  #top-buttons button {
    float: right;
    margin-left: 5px;
  }

  .name-option > td {
    vertical-align: top;
  }
  .name-option.accepted {
    color: #007bff;
    font-size: 20px;
  }
  .decision-text {
    font-size: 11px;
    width: 600px;
    position: relative;
    display: block;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  .search{
    width: 350px;
  }

  #manual-search button {
    width: 20px;
    border: none;
    background-color: transparent;
    padding: 0;
    color: #666;
    margin-top: -1px;
  }

  .btn-search{
    margin-left: -25px;
  }
  .btn-reset {
    margin-left: -40px;
}
</style>

<!-- not scoped -->
<style>
  .name-state-icon .icon-rejected {
    color: #c00;
  }
  .name-state-icon .icon-accepted {
    color: #007bff;
    font-size: 20px;
  }

  .btn-undo {
    padding: 2px 5px;
    font-size: 11px;
  }
</style>
