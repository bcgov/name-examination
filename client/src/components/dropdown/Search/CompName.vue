/* eslint-disable */
<template>
  <div>
    <div class="name-sect">
      Names:<br />
      <div class="row">
        <div class="col name1-font">
          <input id="cmp1" v-model="compName1" onclick="setBorder('cmp1')" class='rtb'>
        </div>
      </div>

      <div class="row">
        <div class="col name2-font">
          <input id="cmp2" v-model="compName2" onclick="setBorder('cmp2')" class='rtb'>
        </div>
      </div>

      <div class="row">
        <div class="col name2-font">
          <input id="cmp3" v-model="compName3" onclick="setBorder('cmp3')" class='rtb'>

        </div>
      </div>

    </div>
  </div>
</template>

<script>
/* eslint-disable */

  export default {
    name: 'CompName',
    computed: {
      compName1() {
        return this.$store.getters.compName1;
      },
      compName2() {
        return this.$store.getters.compName2;
      },
      compName3() {
        return this.$store.getters.compName3;
      }
    },
    mounted() {
    },
    methods: {
      setBorder(id) {
        const tb = document.getElementById(id);
        tb.borderWidth = "1";
      },
      setFocus(id) {
        const ell = document.getElementById(id);
        ell.focus();
      }
    }
  }
</script>


<style scoped>
  .name-sect {
  }
  .name1-font{
    font-size: 2.2em;
    text-align: left;
  }
  .name2-font{
    font-size:1.2em;
    text-align: left;
  }
  .rtb {
    border: 0px;
  }
</style>
