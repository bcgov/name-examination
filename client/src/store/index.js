/* eslint-disable */
import Vue from 'vue'
import Vuex from 'vuex'
import axios from '../axios-auth'
import globalAxios from 'axios'
import router from '@/router'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    nrNum: null,
    authorized: null,
    userId: null,
    email: null,
    kctoken: null,
    nr_conflict: null,
    user_role: null,
    currentChoice: null,
    currentName: null,
    currentState: null,
    is_editing: false,
    nrData: null,
    details: null,
   // formData: {
      compInfo: {
        nrNumber: null,
        compNames: {
          compName1: null,
          compName2: null,
          compName3: null
        },
        requestType: null,
      },
      applicantInfo: {
        applicantname: {
          firstName: null,
          lastname: null
        },
        contactInfo: {
          address: null,
          contactName: null,
          phone: null,
          email: null,
          fax: null
        },
      },
      additionalCompInfo: {
        jurisdiction: null,
        natureOfBussiness: null,
        nuans: null,
        sk_name: null,
        nr_status: null
      },
      examiner: null,
      priority: null,
      reSubmission: {
        reSubmissionYN: null,
        linkedNR: null
      },
      reservationCount: null,
      expiryDate: null,
  //  },
    issueText: null,
    issue: {
        issue_Match: null,
        issue_Consent: null,
        issue_TradeMark: null,
        issue_History: null,
        issue_Format: null,
        issue_Match_Text: null,
        issue_Consent_Text: null,
        issue_TradeMark_Text: null,
        issue_History_Text: null,
        issue_Format_Text: null
      }
  },

  mutations: {
    authUser (state, userData) {
      state.kctoken = userData
      //state.kctoken = userData.client_session
      //state.userRole = userData.user_role
      //state.idToken = userData.client_session
    },

    clearAuthData (state) {
      state.kcToken = null
      state.userId = null
    },

    setIssueText (state,issue) {
      switch (issue) {
        case "M1":
          state.issueText = state.issue.issue_Match_Text;
          break;
        case "M2":
          state.issueText = state.issue.issue_Consent_Text;
          break;
        case "M3":
          state.issueText = state.issue.issue_TradeMark_Text;
          break;
        case "M4":
          state.issueText = state.issue.issue_History_Text;
          break;
        case "M5":
          state.issueText = state.issue.issue_Format_Text;
          break;
        default:
          state.issueText = state.issue.issue_Match_Text;
      }
    },

    loadpostgresNo(state, postgresData) {
        state.compInfo.nrNumber = postgresData.nameRequest
    },

    loadCompanyInfo(state, dbcompanyInfo) {
      console.log('Loading Company Info into state')

      state.nrData = dbcompanyInfo

      if (dbcompanyInfo.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of dbcompanyInfo.names) {
          switch (record.choice) {
            case 1:
              state.compInfo.compNames.compName1 = record.name
              break;
            case 2:
              state.compInfo.compNames.compName2 = record.name
              break;
            case 3:
              state.compInfo.compNames.compName3 = record.name
              break;
            default:
              console.log('Error with company name structure')
              return
          }
        }
      }

      state.compInfo.requestType = dbcompanyInfo.requestTypeCd
      state.applicantInfo.applicantname.firstName = dbcompanyInfo.applicant
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.applicantInfo.contactInfo.contactName = dbcompanyInfo.contact
      state.applicantInfo.contactInfo.phone = dbcompanyInfo.phoneNumber
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.additionalCompInfo.jurisdiction = dbcompanyInfo.xproJurisdiction
      state.additionalCompInfo.natureOfBussiness = dbcompanyInfo.natureBusinessInfo
      state.additionalCompInfo.nuans = dbcompanyInfo.nuansNum
      state.additionalCompInfo.sk_name = dbcompanyInfo.skPartner
      state.additionalCompInfo.nr_status = dbcompanyInfo.state
      state.examiner = dbcompanyInfo.userId
      state.priority = dbcompanyInfo.priorityCD
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.expiryDate = dbcompanyInfo.expiryDate
    },

    loadCompanyIssues(state, dbcompanyIssues) {
      //state.issue.issue_Match = dbcompanyIssues.isue_Match
      //state.issue.issue_Match_Text = dbcompanyIssues.issue_Match_Text
      //state.issue.issue_Consent = dbcompanyIssues.issue_Consent
      //state.issue.issue_Consent_Text = dbcompanyIssues.issue_Consent_Text
      //state.issue.issue_TradeMark = dbcompanyIssues.issue_TradeMark
      //state.issue.issue_TradeMark_Text = dbcompanyIssues.issue_TradeMark_Text
      //state.issue.issue_History = dbcompanyIssues.issue_History
      //state.issue.issue_History_Text = dbcompanyIssues.issue_History_Text
      //state.issue.issue_Format = dbcompanyIssues.issue_Format
      //state.issue.issue_Format_Text = dbcompanyIssues.issue_Format_Text
    },

    update_nrData(state){

      if(state.nrData.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of state.nrData.names) {
          switch (record.choice) {
            case 1:
              state.nrData.names[0].name = state.compInfo.compNames.compName1
              break;
            case 2:
              state.nrData.names[0].name = state.compInfo.compNames.compName2
              break;
            case 3:
              state.nrData.names[0].name = state.compInfo.compNames.compName3
              break;
          }
        }
      }

      state.nrData.requestTypeCd = state.compInfo.requestType
      state.nrData.applicant = state.applicantInfo.applicantname.firstName
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.nrData.contact = state.applicantInfo.contactInfo.contactName
      state.nrData.phoneNumber =  state.applicantInfo.contactInfo.phone
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.nrData.xproJurisdiction = state.additionalCompInfo.jurisdiction
      state.nrData.natureBusinessInfo = state.additionalCompInfo.natureOfBussiness
      state.nrData.nuansNum = state.additionalCompInfo.nuans
      state.nrData.skPartner = state.additionalCompInfo.sk_name
      //state.nrData.state =  state.additionalCompInfo.nr_status
      state.nrData.userId = state.examiner
      state.nrData.priorityCD = state.priority
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.nrData.expiryDate = state.expiryDate
    },

    saveDetail(state,detail){
      state.details = detail
    }
  },

  actions: {
    kcauth ({commit, dispatch, state}) {
      const kc = localStorage.getItem('KEYCLOAK_TOKEN')
      state.kctoken = kc
      //console.log(authData)
      //state.user_role =  authData.realm_access.roles,
      state.user_role = ''
      state.idToken = kc
      localStorage.setItem('kctoken',state.kctoken)
      localStorage.setItem('user_role',state.user_role)

      if (state.kctoken){
        console.log('KC Authorized user roles')
        commit('authUser', {
          //user_role: state.user_role,
          kctoken: state.kctoken
        })
      }
    },

    logout ({commit,state}) {

      commit('clearAuthData')

      localStorage.removeItem('expirationDate')
      localStorage.removeItem('token')
      localStorage.removeItem('user_role')
      localStorage.removeItem('userId')
      localStorage.removeItem('email')
      localStorage.removeItem('COMPINFO')

      localStorage.removeItem('kctoken')
      localStorage.removeItem('KEYCLOAK_TOKEN')
      localStorage.removeItem('AUTHORIZED')
      localStorage.removeItem('USERNAME')

    },

    tryAutoLogin ({commit}) {
    },

    storeUser ({commit, state}, userData) {
      if (!state.idToken) {
        return
      }
      globalAxios.post('/users.json' + '?auth=' + state.idToken, userData)
        .then(res => console.log(res))
        .catch(error => console.log(error))
    },

    fetchUser ({commit, state}) {
      if (!state.idToken) {
        return
      }
      globalAxios.get('/users.json' + '?auth=' + state.idToken)
        .then(res => {
          console.log(res)
          const data = res.data
          const users = []
          for (let key in data) {
            const user = data[key]
            user.id = key
            users.push(user)
          }
          console.log(users)
          commit('storeUser', users[0])
        })
        .catch(error => console.log(error))
    },

    setDetails({commit, state}) {
     var detail = state.details?null:"1";
     commit('saveDetail',detail)
    },

    selectNameIssue ({commit, dispatch, state}, divID) {
      console.log('action: selected issue control:' + divID)
      localStorage.setItem('issueText',divID )
      commit('setIssueText', divID)
    },

    getpostgrescompNo ({commit, dispatch, state}) {
      console.log('action: select next company number from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/queues/@me/oldest'
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp No Response:' + response)
        commit('loadpostgresNo',response.data)
      })
    },

    getpostgrescompInfo ({commit},nrNumber) {
      console.log('action: getting data for company number: ' + nrNumber + ' from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + nrNumber
      console.log('URL:' + url)
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp Info Response:' + response.data)
        localStorage.setItem('COMPINFO',response.data)
        commit('loadCompanyInfo',response.data)
      })
      .catch(error => console.log('ERROR: ' + error))
    },

    updateNRState ({state},nrState) {
      console.log('Updating Examination state for number: ' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber

      axios.patch(url,{headers: {Authorization: `Bearer ${myToken}`}},{"state": nrState} )
           .then(function(response){
                console.log('state updated to ' + nrState + ' for ' + state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
      },

    nameApproved( {commit, state}) {
      console.log('Name Approved for ' + state.compInfo.nrNumber + ", " + state.currentName)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'APPROVED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url, state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name approved for ' + state.compInfo.nrNumber)
           })
           .catch(error => console.log('ERROR: ' + error))
    },

    nameRejected({commit, state}) {
      console.log('Name Rejected for NR #' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'REJECTED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url,state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name rejected for ' +  state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
    }
  },

  getters: {
    is_editing(state) {
      return state.is_editing
    },
    user(state) {
      return state.user
    },
    email(state) {
      return state.email
    },
    issueText(state) {
      return state.issueText
    },
    isAuthenticated(state) {
      return localStorage.getItem("AUTHORIZED")
    },
    nrNumber(state) {
      return state.compInfo.nrNumber
    },
    compName1(state) {
     return state.compInfo.compNames.compName1
    },
    compName2(state) {
      return state.compInfo.compNames.compName2
    },
    compName3(state) {
      return state.compInfo.compNames.compName3
    },
    requestType(state) {
      return state.compInfo.requestType
    },
    firstName(state) {
      return state.applicantInfo.applicantname.firstName
    },
    lastName(state) {
      return state.applicantInfo.applicantname.lastname
    },
    address(state) {
      return state.applicantInfo.contactInfo.address
    },
    conEmail(state) {
      return  state.applicantInfo.contactInfo.email
    },
    contactName(state) {
      return state.applicantInfo.contactInfo.contactName
    },
    phone(state) {
      return state.applicantInfo.contactInfo.phone
    },
    fax(state) {
      return state.applicantInfo.contactInfo.fax
    },
    jurisdiction(state) {
      return  state.additionalCompInfo.jurisdiction
    },
    natureOfBusiness(state) {
      return state.additionalCompInfo.natureOfBussiness
    },
    nuans(state) {
      return state.additionalCompInfo.nuans
    },
    sk_name(state) {
      return state.additionalCompInfo.sk_name
    },
    nr_status(state) {
      return state.additionalCompInfo.nr_status
    },
    examiner(state) {
      return state.examiner
    },
    priority(state) {
      return state.priority
    },
    reSubmissionYN(state) {
      return state.reSubmission.reSubmissionYN
    },
    linkedNR(state) {
      return state.reSubmission.linkedNR
    },
    reservationCount(state) {
      return state.reservationCount
    },
    expiryDate(state) {
      return state.expiryDate
    },
    issue_Match(state) {
      return state.issue.issue_Match
    },
    issue_Match_Text(state) {
      return state.issue.issue_Match_Text
    },
    issue_Consent(state) {
      return state.issue.issue_Consent
    },
    issue_Consent_Text(state) {
      return state.issue.issue_Consent_Text
    },
    issue_TradeMark(state) {
      return state.issue.issue_TradeMark
    },
    issue_TradeMark_Text(state) {
      return state.issue.issue_TradeMark_Text
    },
    issue_History(state) {
      return state.issue.issue_History
    },
    issue_History_Text(state) {
      return state.issue.issue_History_Text
    },
    issue_Format(state) {
      return state.issue.issue_Format
    },
    issue_Format_Text(state) {
      return state.issue.issue_Format_Text
    },
    details(state) {
      return state.details
    },
    addInfo(state) {
      return state.details
    }
  }
})
