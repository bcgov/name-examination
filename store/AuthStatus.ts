import { defineStore } from 'pinia';

export const useAuthStore = defineStore('auth', {
  state: () => ({
    auth: false,
  }),
  actions: {
    toggleAuth() {
      this.auth = !this.auth;
    },
  },
});
