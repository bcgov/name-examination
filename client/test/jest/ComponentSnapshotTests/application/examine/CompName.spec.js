/* eslint-disable */
import Vue from 'vue';
Vue.use(require('vue-shortkey'))
import { shallowMount } from '@vue/test-utils';
import CompName from '@/components/application/Examine/CompName';
import store from '@/store'

describe('CompName.vue ', () => {

    let component;
    beforeEach(() => {
        component = shallowMount(CompName, {store: store});
    });

   it("renders a CompName component", () => {
     expect(component.element).toMatchSnapshot();
   })

  // Add other tests specific to this component and not its sub-components
});
