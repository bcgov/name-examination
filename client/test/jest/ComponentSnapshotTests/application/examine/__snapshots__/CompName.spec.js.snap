// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompName.vue  renders a CompName component 1`] = `
<div>
  <div
    class="name-sect"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          id="top-buttons"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <table>
          <tr
            class="name-option"
          >
            <td>
              1.
            </td>
             
            <td
              id="name1"
            >
              
              
              
              <span
                class="name-state-icon"
              />
               
              <!---->
               
              <span
                class="decision-text"
              >
                
              </span>
            </td>
          </tr>
           
          <tr
            class="name-option"
          >
            <td>
              2.
            </td>
             
            <td
              id="name2"
            >
              
              
              
              <span
                class="name-state-icon"
              />
               
              <!---->
               
              <span
                class="decision-text"
              >
                
              </span>
            </td>
          </tr>
           
          <tr
            class="name-option"
          >
            <td>
              3.
            </td>
             
            <td
              id="name3"
            >
              
              
              
              <span
                class="name-state-icon"
              />
               
              <!---->
               
              <span
                class="decision-text"
                id="decision-text3"
              >
                
              </span>
            </td>
          </tr>
        </table>
         
        <div>
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="modal fade"
    id="add-cancel-comment-modal"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-lg"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
          >
            Please give a comment to explain why this NR is being CANCELLED
          </h5>
           
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
         
        <div
          class="modal-body"
        >
          <textarea
            class="form-control"
            id="cancel-comment-text"
            rows="10"
          />
        </div>
         
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-sm btn-secondary"
            data-dismiss="modal"
            type="button"
          >
            Cancel
          </button>
           
          <button
            class="btn btn-sm btn-danger"
            data-dismiss="modal"
            disabled="disabled"
            id="cancel-nr-after-comment-button"
            type="button"
          >
            CANCEL REQUEST
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
